mlir/utils/jenkins/Jenkinsfile.migraphxintegration

```diff
@@ -0,0 +1,231 @@
+        booleanParam(name: 'fp16', defaultValue: true, description: 'Quantize for fp16')
+        booleanParam(name: 'int8', defaultValue: true, description: 'Quantize for int8')
+        choice(name: 'checkFor', choices:["perf", "verify"], description: 'Do a performance or verification run')
+        string(name: 'mlir_branch', defaultValue: 'develop', description: 'Mlir branch to clone.')
+        string(name: 'migraphx_branch', defaultValue: 'develop', description: 'MIGraphX branch to clone.')
+        booleanParam(name: 'enable_tier1_p0', defaultValue: true, description: 'Enable tests for tier1 P0 models')
+        booleanParam(name: 'enable_tier1_p1', defaultValue: true, description: 'Enable tests for tier1 P1 models')
+        booleanParam(name: 'enable_others', defaultValue: true, description: 'Enable testing for additional models')
+        booleanParam(name: 'MI200', defaultValue: true, description: 'Run the job on a MI210 machine')
+        booleanParam(name: 'Navi3x', defaultValue: true, description: 'Run the job on a Navi3x machine')
+        text(name: 'list_tier1_p0', defaultValue: list_tier1_p0_default, description: 'List of tier 1 models to test (P0)')
+        text(name: 'list_tier1_p1', defaultValue: list_tier1_p1_default, description: 'List of tier 1 models to test (P1)')
+        text(name: 'list_others', defaultValue: '', description: 'List of additional models to test')
+    }
+    stages {
+        stage('BuildAndTest') {
+            matrix {
+                agent any
+                axes {
+                    axis {
+                        name 'PLATFORM'
+                        values "rocm-framework-38.amd.com", "aus-navi3x-13.amd.com"
+                    }
+                }
+                when {
+                    beforeAgent true
+                    expression { return shouldRunOnMachine("${PLATFORM}") }
+                }
+                stages {
+                    stage('BuildAndTest') {
+                        agent {label "${PLATFORM}" }
+                        steps {
+                            echo "Do Build And Test for ${PLATFORM}"
+                            clean()
+                            buildAndTest()
+                            archiveArtifacts allowEmptyArchive: false, artifacts: 'generic.log,tier1_p0.log,tier1_p1.log,other_models.log,summary.log', caseSensitive: true, defaultExcludes: true, fingerprint: false, onlyIfSuccessful: false
+                        }
+                    }
+                }
+            }
+        }
+    }
+    post {
+        success {
+            mail to: "giuseppe.rossini@amd.com,manupa.karunaratne@amd.com,krzysztof.drewniak@amd.com",
+             subject: "MIGraphX integration job passed!",
+             body: "Done"
+        }
+        failure {
+            mail to: "giuseppe.rossini@amd.com,manupa.karunaratne@amd.com,krzysztof.drewniak@amd.com",
+             subject: "Ouch! MIGraphX integration job failed!",
```

Comment:
Could you refactor this into a Jenkins pipeline where you have Jenkins spawn the Docker container?

Reply:
I was thinking to do that after this was merged. The idea of the ticket was to bring the script I wrote to test MIGraphX into our repository. I can refactor it at a later stage (please note that this won't even run in any nightly job for now)

Reply:
Fair enough